components/slider-carousel/src/HeadlessSliderCarousel.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect, useRef, HTMLAttributes, FC } from "react";
+
+interface HeadlessSliderCarouselInterface
+  extends HTMLAttributes<HTMLDivElement> {
+  slides: React.ReactNode[];
+  dependency: number;
+}
+
+const HeadlessSliderCarousel: FC<HeadlessSliderCarouselInterface> = ({
+  slides,
+  dependency,
+  ...rest
+}) => {
+  const parentRef = useRef<HTMLDivElement>(null);
+  let parent: any = parentRef.current;
```

Comment:
Change this from any to HTMLDivElement or related type. Use hover to find the type of ref.